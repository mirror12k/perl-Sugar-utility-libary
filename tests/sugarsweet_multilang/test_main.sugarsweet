
class TestClass {

	sub _constructor(tree args) {
		# self.type = args.type;
		string v = "\t[TestClass] ";
		v += 'hello from multilang!';
		v += " and TestClass!";
		print v;

		self.test_lots_of_messages("pineapple");
		self.test_bag_of_values(20);
		self.test_branch(20);
		self.test_branch(3);
		self.test_branch(5);
		self.test_branch(3);

		foreach (string item in ['lettera', 'letterb', 'letterc', 'letterd']) {
			var msg = self.test_get_message({
				type => item,
			});

			if (msg) {
				print msg;
			} else {
				print "\t[TestClass] got no message";
			}
		}

		list items = [];
		list push items, [ "\t[TestClass] lol" ];
		list push items, [ "5", "6", "7" ];
		list push items, [ "asdf", "qwer" ];
		print join "+", items;


		list a = ["\t[TestClass] zeroth", 'first'];

		list b = a;

		push b, 'second';

		list push b, [ 'third', 'fourth' ];
		print join "+", a;
	}

	void sub test_lots_of_messages(string thing) {
		int i = 1;
		while (i < 6) {
			print "\t[TestClass] my thing #$i is: $thing";
			i += 1;
		}
	}

	void sub test_bag_of_values(int count) {
		int i = 0;
		list vals = [];
		while (i < count) {
			push vals, "$i";
			i += 1;
		}
		string vals_string = join ',', vals;

		print "\t[TestClass] my vals: $vals_string";
	}

	void sub test_branch(int val) {
		if (val < 5) {
			print "\t[TestClass] $val is less than 5";
		} elsif (val > 5) {
			print "\t[TestClass] $val is more than 5";
		} else {
			print "\t[TestClass] $val is 5";
		}
	}

	string sub test_get_message(tree statement) {
		string switch (statement['type']) {
			"lettera": {
				return "\t[TestClass] got message of apple";
			}
			"letterb": {
				return "\t[TestClass] got message of banana";
			}
			"letterc": {
				return "\t[TestClass] got message of cherry";
			}
			default: {
				return;
			}
		}
	}

	# main
	void sub main () native:perl5 {{
		my $f = TestClass->new({
			message => 'hello from perl5!',
		});
	}}

	void sub main () native:php7 {{
		$f = new TestClass([
			'message' => 'hello from php7!',
		]);
	}}

	void sub main () native:python3 {{
		f = TestClass({
			'message':'hello from python3!',
		})
	}}

	void sub main () native:javascript6 {{
		var f = new TestClass({
			'message':'hello from javascript6!',
		})
	}}

	void sub main () native:csharp {{
		new TestClass(new Dictionary<string, object>{
			{ "message", "hello from C#!" },
		});
	}}

}

