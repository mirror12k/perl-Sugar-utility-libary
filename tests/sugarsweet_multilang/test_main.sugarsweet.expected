	[TestClass] hello from multilang! and TestClass!
	[TestClass] my thing #1 is: pineapple
	[TestClass] my thing #2 is: pineapple
	[TestClass] my thing #3 is: pineapple
	[TestClass] my thing #4 is: pineapple
	[TestClass] my thing #5 is: pineapple
	[TestClass] my vals: 0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19
	[TestClass] 20 is more than 5
	[TestClass] 3 is less than 5
	[TestClass] 5 is 5
	[TestClass] 3 is less than 5
	[TestClass] got message of apple
	[TestClass] got message of banana
	[TestClass] got message of cherry
	[TestClass] got no message
	[TestClass] lol+5+6+7+asdf+qwer
	[TestClass] zeroth+first+second+third+fourth
